doc/dev/communication.rst

```diff
@@ -0,0 +1,251 @@
+.. _communication:
+
+Communication in Arbor
+======================
+
+Communication between cells (and thus cell groups) is facilitated using discrete
+events which we call `spikes` in analogy to the underlying neurobiological
+process. Spikes are the only way to communicate between different cell kinds, eg
+cable cells and integrate-and-fire. In accordance to current theory, spikes are
+reduced to a simple time value. The connection is abstracted into a weight and a
+delay; modelling all axonal processes. While this may seem crude, it is a well
+supported model and commonly used in neuronal simulations.
+
+Connections are formed between sources (cable cell: spike detectors) and targets
+(cable cell: synapses). During runtime, events from all sources are concatenated
+on all MPI ranks using ``Allgatherv`` and targets are responsible for selecting
+events they have subscribed to. This is optimised for by sorting events locally
+(by source) and relying on the process layout to convert this into a *globally*
+sorted array.
```

Comment:
That's not really part of the communication. It's also explained in the dev section

Reply:
> Maybe it is helpful to clarify that a `arbor::communicator` holds this state (the concrete connection table, even if it's not a table) in Arbor, even when not using MPI (because the word communicator sounds MPI-ish).

That's explained in the second heading. The information here is starting with the goal (communicating spikes) and then working backwards from there (what's a communicator, how to build the tables, ...).

Reply:
The second heading (Exchange of Spikes) shows an implementation for `communicator::exchange`, not a description of a `communicator`. The struct is shown much further down, line 205. I think it's easier to follow this text if people are aware of what it contains up front.

Reply:
I'd rather not do that upfront, it's way too much to take in. The reasoning is that `exchange` is what people
are looking for and `communicator`'s primary job. Thus the class is built around it and we explain that part 
first. The surrounding and supporting infrastructure and how it's set up comes later.